hypershift-operator/controllers/nodepool/metrics/metrics.go

```diff
@@ -34,6 +34,12 @@ var (
 		Help: "Time in seconds it took from initial NodePool creation and rollout of initial version",
 		Name: NodePoolInitialRolloutDurationMetricName,
 	}, labelNames)
+
+	NodePoolUpgradeDurationMetricName = "hypershift_nodepools_upgrade_duration_seconds"
+	nodePoolUpgradeDuration           = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Help: "Time in seconds it took a NodePool to upgrade and rollout a given version",
+		Name: NodePoolUpgradeDurationMetricName,
```

Comment:
Can you elaborate which value does this metric provide? fwiw This value is directly impacted by the number of replicas of the NodePool and the maxUnavailble/maxSurge values